test/common/api_utils.py

```diff
@@ -143,7 +143,85 @@ def api_node_select(client,
             print '**** Empty node ID list.\n'
     return node_id_list
 
+def api_valivate_node_pollers(client, node_id_list, all_pollers=False):
+    """
+    This function validates that the presented list of node contains active pollers. By default
+    only the fist poller of the pollers assocaited with a node is validates. Setting the
+    "all_pollers" flag to True allows all poller assocaited with a node to be validated.
+    :param  client: reference to config.api2_0_config
+    :param  node_id_list - list of node id's
+    :param  type - type of node to be used (default: 'compute')     #TODO if no node_list, collect all nodes
+    :param  all_pollers - Test all poller assocaited with a node (default: False)
+    return: True  - all nodes have active poller
+            False - all nodes do not have active pollers
+    """
+    pollers_list = []
+    good_pollers = True
+    # TODO: this will change in the near future
+    verbosity = int(os.getenv("VERBOSITY", "0"))
+
+    if not isinstance(node_id_list, (tuple, list, set)):
+        print '**** Provided node_list is not a [list]'
+        # return no good pollers
+        return False
+
+    for node_ld in node_id_list:
+        Api().nodes_get_pollers_by_id(node_ld)
+        node_poller_rsp = client.last_response
+        node_poller_list = loads(node_poller_rsp.data)
+        if node_poller_rsp.status != 200:
+            print '**** Unable to retrieve node: {} pollers via API. status ({})\n'.format(node.get('id'),
+                                                                                           node_poller_rsp.status)
+            # unable to get poller for this node, so return a False
+            return False
+
+        # if the poller return None or an empty list, return a null list
+        if not node_poller_list:
+            print 'Node {} has no pollers.'.format(node_ld)
+            # no poller on this node, so return a False
+            return False
+
+        for poller in node_poller_list:
+            pollers_list.append(poller)
+            if not all_pollers:
```

Comment:
maybe add a comment here that we are only returning the first poller if all_pollers is not true.  You explained it above, but I forgot mid review and this logic kinda threw me for a minute.